docs/multicall-notes.md

```diff
@@ -0,0 +1,225 @@
+| feeRecipient | `0x0000000000000000000000000000000000000000` |
+| mixHash | `0x0000000000000000000000000000000000000000000000000000000000000000` |
+| nonce | `0x0` |
+| extraData | `0x0000000000000000000000000000000000000000000000000000000000000000` |
+| difficulty | The same as the base block defined as the second parameter in the call |
+| gasLimit | The same as the base block defined as the second parameter in the call |
+| hash | Calculated normally, except for phantom blocks, see below Phantom block section |
+| parentHash | Previous blocks hash (the real hash, or phantom blocks hash) |
+| timestamp | The timestamp of previous block + 1 |
+| baseFeePerGas | Calculated on what it should be according to ethereum's spec. Note: baseFeePerGas is not adjusted in the phantom blocks. |
+| sha3Uncles | Empty trie root |
+| withdrawals | Empty array |
+| uncles | Empty array |
+| number | Previous block number + 1 |
+| logsBloom | Calculated normally. ETH logs are not part of the calculation |
+| receiptsRoot | Calculated normally |
+| transactionsRoot | Calculated normally |
+| size | Calculated normally |
+| withdrawalsRoot | Calculated normally |
+| gasUsed | Calculated normally |
+| stateRoot | Calculated normally |
+
+An interesting note here is that we decide timestamp as `previous block timestamp + 1`, while `previous block timestamp + 12` could also be an assumed default. The reasoning to use `+1` is that it's the minimum amount we have to increase the timestamp to keep them valid. While `+12` is what Mainnet uses, there are other chains that use some other values, and we didn't want to complicate the specification to consider all networks.
+
+### Phantom blocks
+The multicall allows you to define on what block number your calls or transactions are being executed on. E.g, consider following call:
+```json
+{
+	"jsonrpc": "2.0",
+	"id": 1,
+	"method": "eth_multicallV1",
+	"params": [
+		{
+			"blockStateCalls": [
+				{
+					"blockOverrides": {
+						"number": "0x64"
+					},
+				},
+				{
+					"blockOverrides": {
+						"number": "0xc8"
+					},
+				}
+			]
+		},
+		"0xa"
+	]
+}
+```
+
+Here we want our calls to be executed in blocks 100 (`0x64`) and in 200 (`0xc8`). The block numbers can be anything as long as they are increasing and higher than the block we are building from 10 (`0xa`). Now we end up in a situation where there exists block ranges 13-99 and 101-199 that are not defined anywhere. These blocks are called "phantom blocks". What happens if you try to request block hash of any of such blocks in the EVM? How can we calculate the block hash of future blocks when we don't know the block hash of the previous block?
+
+Our solution to this problem is to define block hash of a phantom block to be:
+
+```
+keccak(rlp([hash_of_previous_non_phantom_block, phantom_block_number]))
+```
+
+So for example in our example, you could get block hash of block 142 as follows: 
+```
+keccac(rlp([hash of block 12, 142]))
```

Comment:
```suggestion
keccak(rlp([hash of block 12, 142]))
```

Reply:
Also, shouldn't it be hash of block 10? The second param of the call is block `0xa` (so, block 10). So block 10 is the `previous_non_phantom_block`, so it should be 10 right? (not block 12)